lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+    // it helps avoid disallowTrailingWhitespace errors on last line
+    lines.push(/\S/.test(lastLine) ? lastLine : '');
+
+    // calculate min line offset
+    offset = Math.min.apply(null, lines.map(function(line) {
+        // skip empty lines
+        if (!line) {
+            return Infinity;
+        }
+
+        // fetch whitespaces at the line begining
+        var offsetStr = line.match(/^\s*/)[0];
+        var tabCount = offsetStr.match(/\t*/)[0].length;
+
+        // mixed spaces and tabs in one offset -> don't remove offsets
+        if (tabCount && tabCount !== offsetStr.length) {
+            return 0;
+        }
+
+        if (tabCount) {
+            if (spaceOnlyOffset) {
+                // no spaces, but previous offset has ony spaces -> mixed spaces and tabs
+                return 0;
+            } else {
+                // remember offset contains only tabs
+                tabOnlyOffset = true;
+            }
+        } else {
+            if (tabOnlyOffset) {
+                // no tabs, but previous offset has only tabs -> mixed spaces and tabs
+                return 0;
+            } else {
+                // remember offset contains only spaces
+                spaceOnlyOffset = true;
+            }
+        }
+
+        return offsetStr.length;
+    }));
+
+    // remove common offsets if possible
+    if (offset) {
+        lines = lines.map(function(line) {
+            return line.substr(offset);
+        });
+    }
+
+    return {
+        source: lines.join('\n'),
+        offset: offset,
+        lineCount: lineCount
+    };
+}
+
+/**
+ * Parse HTML and search for <script> sources. Each script source also normalize
+ * by line offset. Result contains script sources with information about line
+ * offset (that was removed for each line) and lines count before script source.
+ * This information helps restore absolute positions in html file for errors.
+ * @param {String}
+ * @returns {Object[]}
+ */
+function extractJs(filename, data) {
+    var errors = new Errors(new HtmlFile(filename, data));
+    var scripts = getScripts(data);
+    var sources = [];
+    var line = 1;
+    var lastHtmlPos = 0;
+
+    for (var i = 0; i < scripts.length; i++) {
+        // fetch script source and normalize it
+        var scriptInfo = scripts[i];
+        var normalized = normalizeSource(scriptInfo.source);
+
+        // add line offset before script
+        line += data.substring(lastHtmlPos, scriptInfo.start).split(/\r\n?|\n\r?/).length - 1;
+
+        // store result
+        sources.push({
+            source: normalized.source,
+            offset: normalized.offset,
+            line: line
+        });
+
+        // save offsets for next fragment
+        line += normalized.lineCount - 1;
+        lastHtmlPos = scriptInfo.end;
+    }
+
+    return {
+        sources: sources,
+        errors: errors,
+        addError: function(error) {
+            errors._errorList.push({
```

Comment:
It doesn't work as you expected https://github.com/jscs-dev/node-jscs/blob/master/lib/errors.js#L34
I'll be happy if it does